### PR TITLE
Add support for node/iojs/nw/electron mirrors

### PR DESCRIPTION
**Pull request by w1ndy**
This pull request proposes the following changes:

* adding a new argument ``-M, --mirrors`` to supply mirrors for downloading dist files because some regions cannot access official Node.js website fast enough. The syntax of this argument is like ``-M node=http://nodejs.org/dist,iojs=https://iojs.org/dist``.
* updating the corresponding tests

### CODE REVIEW

**Comment by unbornchikken** [comment]
So, the idea is right but the solution is not so. 

If you publish your module with a package.json referencing custom mirrors in its install script's arguments, then users of your module will use that mirror no matter what, because npm install will invoke your script as is.

The parameter is good as is, but we should support and advertise an use case that relies on npm config keys instead.

File: bin/cmake-js [hunk 1]
```diff
@@ -4,6 +4,7 @@
 var log = require("npmlog");
 var _ = require("lodash");
 var BuildSystem = require("../").BuildSystem;
+var Environment = require("../").environment
```
Comment (unbornchikken) [inline bin/cmake-js]: Why capitalize Environment there?

File: bin/cmake-js [hunk 2]
```diff
@@ -165,6 +171,26 @@ _.keys(argv).forEach(function (key) {
     }
 });
 
+function parseMirrors(mirrors) {
+    var result = {};
+    var validMirrorKeys = ["node", "iojs", "nw", "electron"]
```
Comment (unbornchikken) [inline bin/cmake-js]: ; missing

File: bin/cmake-js [hunk 3]
```diff
@@ -165,6 +171,26 @@ _.keys(argv).forEach(function (key) {
     }
 });
 
+function parseMirrors(mirrors) {
```
Comment (unbornchikken) [inline bin/cmake-js]: iojs is not supported atm by CMake.js

**Comment by Julusian** [comment]
Since this PR was opened, it is possible to configure these through environment variables https://github.com/cmake-js/cmake-js/blob/master/lib/runtimePaths.js#L6-L7

**Pull request closed without merging**